### ISSUE TITLE
rack 3.x compatibility

### ISSUE THREAD

**Issue by gsar** [axsuul/rails-reverse-proxy#75]
seeing deprecation warnings with rack 3.0.9.

```
2024-02-13 18:36:09 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:09 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:09 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:18 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:18 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:18 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:29 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:29 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:36:29 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:39:45 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:39:45 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
2024-02-13 18:39:45 /usr/local/bundle/3.2.3/gems/rails-reverse-proxy-0.12.0/lib/reverse_proxy/client.rb:135: warning: Rack::Utils::HeaderHash is deprecated and will be removed in Rack 3.1, switch to Rack::Headers
```

**Comment by davidrunger** [comment]
#76 fixes this.

[**Edit:** There is no need to install from this GitHub commit anymore; this fix has been released as version `0.13.0`. 👍] ~~To start using that now, you can switch your `rails-reverse-proxy` listing in your `Gemfile` to the following (though, **caution** 🚨, I might delete that fork at any time (particularly if/when that PR is merged), at which point I guess that this will stop working, although then you should switch back to the RubyGems-released version of `rails-reverse-proxy'`, anyway):~~

```rb
gem 'rails-reverse-proxy',
  # https://github.com/axsuul/rails-reverse-proxy/pull/76
  github: 'davidrunger/rails-reverse-proxy',
  ref: 'f801620d2c7b94bc7fbecebe3cdbae46aa021656'
```

**Comment by axsuul** [comment]
Fixes have been released as `0.13.0`. Thanks!
